force-app/main/default/classes/TestRunProcessor.cls

```diff
@@ -108,6 +108,46 @@ public with sharing class TestRunProcessor implements Schedulable {
             );
         }
 
-        insert results;
+        List<Test_Run_Method_Result__c> result = processPassFailDates(results);
```

Comment:
result === results

This assignment effectively does nothing as `processPassFailDates` returns the same list that was passed in